README.md

```diff
@@ -1,583 +1,317 @@
-        ports:
-        - containerPort: 8080
+```
+go install github.com/google/ko
 ```
 
-It is notable that this is not the default (anymore) because certain popular
-registries (including Docker Hub) do not support multi-level repository names.
+Or, to install at a specific version:
 
-`ko resolve`, `ko apply`, and `ko create` accept an optional `--selector` or
-`-l` flag, similar to `kubectl`, which can be used to filter the resources from
-the input Kubernetes YAMLs by their `metadata.labels`.
+```
+go install github.com/google/ko@v0.7.0
+```
 
-In the case of `ko resolve`, `--selector` will render only the resources that
-are selected by the provided selector.
+## Authenticate
 
-See
-[the documentation on Kubernetes selectors](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/)
-for more information on using label selectors.
+`ko` depends on the authentication configured in your Docker config (typically `~/.docker/config.json`). If you can push an image with `docker push`, you are already authenticated for `ko`.
 
-### `ko apply`
+Since `ko` doesn't require `docker`, `ko login` also provides a surface for logging in to a container image registry with a username and password, similar to [`docker login`](https://docs.docker.com/engine/reference/commandline/login/).
 
-`ko apply` is intended to parallel `kubectl apply`, but acts on the same
-resolved output as `ko resolve` emits. It is expected that `ko apply` will act
-as the vehicle for rapid iteration during development. As changes are made to a
-particular application, you can run: `ko apply -f unit.yaml` to rapidly rebuild,
-repush, and redeploy their changes.
+## Choose Destination
 
-`ko apply` will invoke `kubectl apply` under the hood, and therefore apply to
-whatever `kubectl` context is active.
+`ko` depends on an environment variable, `KO_DOCKER_REPO`, to identify where it should push images that it builds. Typically this will be a remote registry, e.g.:
 
-### `ko apply --watch` (EXPERIMENTAL)
+- `KO_DOCKER_REPO=gcr.io/my-project`, or
+- `KO_DOCKER_REPO=my-dockerhub-user`
 
-The `--watch` flag (`-W` for short) does an initial `apply` as above, but as it
-does, it builds up a dependency graph of your program and starts to continuously
-monitor the filesystem for changes. When a file changes, it re-applies any yamls
-that are affected.
+# Build an Image
 
-For example, if I edit `github.com/foo/bar/pkg/baz/blah.go`, the tool sees that
-the `github.com/foo/bar/pkg/baz` package has changed, and perhaps both
-`github.com/foo/bar/cmd/one` and `github.com/foo/bar/cmd/two` consume that
-library and were referenced by `config/one-deploy.yaml` and
-`config/two-deploy.yaml`. The edit would effectively result in a re-application
-of:
+`ko publish ./cmd/app` builds and pushes a container image, and prints the resulting image digest to stdout.
 
 ```
-ko apply -f config/one-deploy.yaml -f config/two-deploy.yaml
+$ ko publish ./cmd/app
```

Comment:
We should use `$` (or not) consistently.

Reply:
Yeah that's fair. I tend to like including the `$` when later lines include output, to denote which was input vs output, but it's also annoying to be inconsistent. Dropping the `$`.